### PR TITLE
Feature#14 add styles to feature list all employees

### LINKED ISSUE #14: Add styles to feature list all employees
# REQUIREMENTS
In this issue all styles are going to be added in the views 

# DESIGN
Provide references to images of the design, how it looks like now, and how it should look like.

## View a new employee
Pending design

## List all employee
imagen

## View of a specific employee
imagen

# Acceptance Criteria
List all required acceptance criteria
- [x] The final result is like the design

### PR DESCRIPTION
**Pull request by ArielMD**
I added styles to the Employee's CRUD
 
Using Bootstrap and Sass, I added the styles to make them similar to design

## Description: Why did you change/implement that?

To give a better UI to the project

## Testing: Help me how can I test or look at the changes?

you can see this changes in the images below

image
image
image

### CODE REVIEW

File: reloj-checador/app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
+    <link rel="stylesheet" href="https://pro.fontawesome.com/releases/v5.10.0/css/all.css" integrity="sha384-AYmEC3Yw5cVb3ZcuHtOA93w35dYTsvhLPVnYs9eStHfGJvOvKxVfELGroGkvsg+p" crossorigin="anonymous"/>
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/layouts/application.html.erb]: WDYT about adding the fontawesome assets via webpack?

1. console:
```bash
yarn add @fortawesome/fontawesome-free
```

2. import it in the javascript/packs/application.js:
```javascript
import "@fortawesome/fontawesome-free/css/all"
```

3. Check if it works:
```html
<i class="far fa-address-book"></i>
<i class="fab fa-github"></i>
````
That’s it!😊

File: reloj-checador/app/assets/stylesheets/employees.scss [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+    width: 80%;
+    margin: auto;
+}
+
+.title{
+    text-align: center;
+    margin: 2rem;
+}
+
+table{
```
Comment (EduardoGHdez) [inline reloj-checador/app/assets/stylesheets/employees.scss]: I'd suggest creating a CSS class rather than styling directly the HTML elements

File: reloj-checador/app/assets/stylesheets/scaffolds.scss [hunk 3]
```diff
@@ -1,65 +1,58 @@
...
-//   background-color: #c00;
-//   color: #fff; }
-
-// #error_explanation ul li {
-//   font-size: 12px;
-//   list-style: square; }
-
-// label {
-//   display: block; }
+*{
```
Comment (EduardoGHdez) [inline reloj-checador/app/assets/stylesheets/scaffolds.scss]: Do we really want to apply these styles to every single element in the DOM?
Why is this necessary?

File: reloj-checador/app/assets/stylesheets/scaffolds.scss [hunk 4]
```diff
@@ -1,65 +1,58 @@
...
+    padding: 0px;
+    margin: 0px;
+    box-sizing: border-box;
+}
+
+body {
```
Comment (EduardoGHdez) [inline reloj-checador/app/assets/stylesheets/scaffolds.scss]: Same as above, regarding styling HTML elements

File: reloj-checador/app/views/employees/index.html.erb [hunk 5]
```diff
@@ -1,39 +1,51 @@
...
+              <%= link_to employee, method: :delete, data: { confirm: 'Are you sure?' }, :class => "btn btn-destroy" do %>
+                <i class="fas fa-trash"></i>Destroy
+              <% end %>
+            </td>
+          </tr>
+        <% end %>
+      </tbody>
+    </table>
+  </div>
+</div>
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/employees/index.html.erb]: 🔥

File: reloj-checador/app/views/employees/index.html.erb [hunk 6]
```diff
@@ -1,39 +1,51 @@
...
+                <i class="fas fa-user-edit"></i>Edit
+              <% end %>
+            </td>
+            <td>
+              <%= link_to employee, method: :delete, data: { confirm: 'Are you sure?' }, :class => "btn btn-destroy" do %>
+                <i class="fas fa-trash"></i>Destroy
+              <% end %>
+            </td>
+          </tr>
+        <% end %>
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/employees/index.html.erb]: There is no need of changing anything here but just wanted to let you know that rails provide a mechanism for creating partial views. And with those, we are able to just pass a collection/list/array of elements, and it'll render just as you are doing it here.

Mentioning it because it's so common found partials-usage in rails projects 😄 

Rails Guides/ Rendering Collections: https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-collections

File: reloj-checador/app/views/employees/_form.html.erb [hunk 7]
```diff
@@ -10,44 +12,59 @@
...
+      <small class="message-error"><%= employee.errors[:private_number].join(" ")%></small>
+    <%end%>
   </div>
 
-  <div class="field">
-    <%= form.label :active %>
-    <%= form.check_box :active %>
+  <div class="form-group">
+    <%= form.label :company, :class => "inputState"%>
+    <%= select("employee", "company_id", Company.all.collect {|c| [c.name, c.id]}, {include_blank: "Select a company"},{:class => "form-control"}) %>
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/employees/_form.html.erb:45]: Try to avoid introducing logic in your views. 

By any chance can we move the `Company` query to the proper controller?

File: reloj-checador/app/views/employees/show.html.erb [hunk 8]
```diff
@@ -1,40 +1,82 @@
...
+          </tr>
+          <tr>
+            <th scope="row">3</th>
+            <td>Larry</td>
+            <td>the Bird</td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/employees/show.html.erb]: Why is this hard-coded attendance list needed?

Do you have plans for updating it to make it dynamic?

**Review by EduardoGHdez — DISMISSED** [review]
OMG! OMG! OMG! 🔥 

Congratulations!!!!!
What a nice enhancement
And I love you kept your git-history super clean 💯 🎉 

Love it, love it, love it! :heart:

I'd just say that the `feat:` prefix is not needed at all in your commit messages. Just by having a clear title of what we are attempting to do is enough. 

I'm not sure if anyone else had shared this resource with you before. But wanted to share it, because I'd like to help you to continue improving and growing! 

It is a classic! 😂  https://chris.beams.io/posts/git-commit/

File: reloj-checador/app/views/employees/_form.html.erb [hunk 9]
```diff
@@ -1,4 +1,6 @@
 <%= form_with(model: employee) do |form| %>
+
+<!--
```
Comment (EduardoGHdez) [inline reloj-checador/app/views/employees/_form.html.erb]: I think this is not needed he he

File: reloj-checador/app/assets/stylesheets/employees.scss (hunk 2 above)
Comment (EduardoGHdez) [reply reloj-checador/app/assets/stylesheets/employees.scss]: NVM, just noticed you removed it later in the PR

File: reloj-checador/app/assets/stylesheets/scaffolds.scss (hunk 3 above)
Comment (EduardoGHdez) [reply reloj-checador/app/assets/stylesheets/scaffolds.scss]: Same as above

File: reloj-checador/app/views/layouts/application.html.erb (hunk 1 above)
Comment (ArielMD) [reply reloj-checador/app/views/layouts/application.html.erb]: I didn't know that, I'm going to try xd

File: reloj-checador/app/views/employees/show.html.erb (hunk 8 above)
Comment (ArielMD) [reply reloj-checador/app/views/employees/show.html.erb]: yes i do, but feature attendance wasn´t implemented in this brach, so i only did the styles, later we going to implements the dinamic data

File: reloj-checador/app/views/employees/_form.html.erb (hunk 9 above)
Comment (ArielMD) [reply reloj-checador/app/views/employees/_form.html.erb]: I did not realize that xd, thaks for the resources
